packages/dds/matrix/src/matrix.ts

```diff
@@ -624,25 +820,50 @@ export class SharedMatrix<T = any>
 				this.rows.applyMsg(msg, local);
 				break;
 			default: {
+				if (contents.type === MatrixOp.changeSetCellPolicy) {
+					this.setCellPolicySwitchOpSeqNumber = rawMessage.sequenceNumber;
+					return;
+				}
+
 				assert(
 					contents.type === MatrixOp.set,
 					0x021 /* "SharedMatrix message contents have unexpected type!" */,
 				);
 
-				const { row, col } = contents;
-
+				const { row, col, value } = contents;
+				assert(rawMessage.clientId !== null, "clientId should not be null!!");
 				if (local) {
 					// We are receiving the ACK for a local pending set operation.
 					const { rowHandle, colHandle, localSeq } = localOpMetadata as ISetOpMetadata;
-
-					// If this is the most recent write to the cell by the local client, remove our
-					// entry from 'pendingCliSeqs' to resume allowing remote writes.
-					if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
-						this.pending.setCell(rowHandle, colHandle, undefined);
+					if (this.isSetCellOpMadeInFWWPolicy(rawMessage)) {
+						if (this.shouldSetCellBasedOnFWW(rowHandle, colHandle, rawMessage)) {
+							this.cellLastWriteTracker.setCell(rowHandle, colHandle, {
+								seqNum: rawMessage.sequenceNumber,
+								clientId: rawMessage.clientId,
+							});
+						} else {
+							// conflict. We would already set the right contents in the cell, when we would have received
+							// set op for that cell from other client.
+							this.emit(
+								"conflict",
+								row,
+								col,
+								this.cells.getCell(rowHandle, colHandle), // Current value
+								value, // Ignored value
+								this,
+							);
+						}
+					} else {
+						if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
+							this.pending.setCell(rowHandle, colHandle, undefined);
```

Comment:
Same comment here about isLatestPendingWrite() as in resubmit flow.

Reply:
Fixed the resubmitFlow. See the comments/code there. Here we should not apply intermediate states as for this local client if we apply all intermediate states, then if this local client reads value, it will not get the latest state which it has just set and knows that it should be the latest state only. Plus, it is optimized also.